modules/uncertainty/R/get.parameter.samples.R

```diff
@@ -106,13 +111,16 @@ get.parameter.samples <- function(settings,
     
     ### When no ma for a trait, sample from prior
     ### Trim all chains to shortest mcmc chain, else 20000 samples
-    if(!is.null(distns$prior.distns)){
-      priors <- rownames(distns$prior.distns)
-    } else {
-      priors <- NULL
-    }  
-    if (!is.null(distns$trait.mcmc)) {
-      param.names[[i]] <- names(distns$trait.mcmc)
+    if(exists("post.distns") & exists("trait.mcmc")){
+      priors <- c(names(trait.mcmc), rownames(post.distns))
+      prior.distns <- post.distns
+    }else{
+      priors <- rownames(prior.distns)
+    }
+    #priors <- names(trait.mcmc)
+  
+    if (exists("trait.mcmc")) {
+      param.names[[i]] <- names(trait.mcmc)
```

Comment:
Has the above change been well tested in the different cases where this function is used (e.g., ensemble runs, sensitivity analyses, PDA, SDA) and from the different possible choices of parameter inputs (e.g. running from uninformative priors, running from trait meta-analysis, and running from PDA posteriors)? Or at a minimum, have you thought through the logic of all those cases to make sure you're not changing intended functionality?  I'm concerned that changes made in one context might break the code (or lead to unintended consequences) in others.